engine/src/available-command.ts

```diff
@@ -445,29 +480,32 @@ export function possibleFreeActions(engine: Engine, player: Player) {
   if (pl.data.burnablePower() > 0) {
     commands.push({
       name: Command.BurnPower,
-      player,
+      player: pl.player,
       data: range(1, pl.data.burnablePower() + 1),
     });
   }
 
   return commands;
 }
 
-function transformToSpendCommand(actions: { cost: string; income: string }[], player: PlayerObject) {
-  const acts = [];
-  for (const freeAction of actions) {
-    const maxPay = player.maxPayRange(Reward.parse(freeAction.cost));
+export function freeActionData(availableFreeActions: FreeAction[], player: PlayerObject): AvailableFreeAction[] {
+  const acts: AvailableFreeAction[] = [];
+  for (const freeAction of availableFreeActions) {
+    const maxPay = player.maxPayRange(Reward.parse(freeActionConversions[freeAction].cost));
     if (maxPay > 0) {
       acts.push({
-        cost: freeAction.cost,
-        income: freeAction.income,
```

Comment:
Those should still be forwarded. `action` was added for the sake of shortcuts. A bot / less sophisticated viewer or client would need `cost` and `income`.